### PR TITLE
feat: bump protoc and protobuf version

### PR DESCRIPTION
**Pull request by JoanFM**
Goals:

- ...
- ...
- [ ] check and update documentation. See [guide](https://github.com/jina-ai/jina/CONTRIBUTING.md#documentation-guidelines) and ask the team.

### CODE REVIEW

File: Dockerfiles/protogen.Dockerfile [hunk 1]
```diff
@@ -5,7 +5,7 @@ RUN apt-get update && apt-get install --no-install-recommends -y
 WORKDIR /builder/
 
 RUN apt-get install --no-install-recommends -y build-essential git autoconf libtool wget unzip zlib1g-dev pkg-config cmake
-RUN wget https://github.com/protocolbuffers/protobuf/releases/download/v3.19.1/protoc-3.19.1-linux-x86_64.zip  -O protobuf.zip \
+RUN wget https://github.com/protocolbuffers/protobuf/releases/download/v3.20.1/protoc-3.20.1-linux-x86_64.zip  -O protobuf.zip \
```
Comment (hanxiao) [inline Dockerfiles/protogen.Dockerfile:8]:
```suggestion
RUN wget https://github.com/protocolbuffers/protobuf/releases/download/v21.0/protoc-21.0-linux-x86_64.zip  -O protobuf.zip \
```

maybe directly to 21 as i see 21 introduces a lot of improvements

File: jina/resources/extra-requirements.txt [hunk 2]
```diff
@@ -26,7 +26,7 @@
 # demo: required for the hello world demos
 
 numpy:                      core
-protobuf>=3.19.1,<=3.20.1:  core
+protobuf>=3.21.0:           core
```
Comment (nan-wang) [inline jina/resources/extra-requirements.txt:29]: https://pypi.org/project/protobuf/4.21.0/

```suggestion
protobuf>=4.21.0:           core
```

**Review by nan-wang — CHANGES_REQUESTED** [review]

**Review by Johnson990 — APPROVED** [review]

File: Dockerfiles/protogen.Dockerfile (hunk 1 above)
Comment (Johnson990) [reply Dockerfiles/protogen.Dockerfile]: Hacked

**Review by samsja19 — APPROVED** [review]

**Comment by samsja** [comment]
closing this PR has the bump in protobuf version has been handled in this PR: https://github.com/jina-ai/jina/pull/5082

**Comment by samsja** [comment]
closing this PR has the bump in protobuf version has been handled in this PR: https://github.com/jina-ai/jina/pull/5082

**Pull request closed without merging**